Cargo.toml

```diff
@@ -15,6 +15,11 @@ build = "build.rs"
 # Supports tracking channels with a non-bitcoin chain hashes. Currently enables all kinds of fun DoS attacks.
 non_bitcoin_chain_hash_routing = []
 fuzztarget = ["secp256k1/fuzztarget", "bitcoin/fuzztarget"]
+max_level_error = []
```

Comment:
My point about overhead was more a question of "would someone care about the peroformance hit of this" - the reason to have the conditional compilation is that the act of doing the format!() itself can often be nontrivial, especially when you're running at something like trace, *but* because rust lets us do format_args!() without actually doing the full format!() unless the client actually wants to write out the line to their log, the overhead of doing a log call that gets dropped should be pretty low (mostly the indirect call through the Arc, but given we're doing a ton of crypto anyway so what's some indirect calls, which theoretically we could get dropped by templating if we needed to in the future). All that said, having the options isn't many lines of code, so I don't feel so strongly anymore.

Still, if we do want to keep the features, I'd think its more intuitive to have them go the other way - default to call the user's Logger for everything TRACE and up, and let the user turn things off at compile-time if they want to opt into it.